### PR TITLE
Homework 4

### PR DESCRIPTION
**Pull request by zkolesnyk**

### CODE REVIEW

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private static final Random random = new Random();
+    private static final Shape[] shapes = Shape.values();
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(shapes.length);
+        String figureName = shapes[randomNumber].name();
+
+        switch (figureName) {
+            case "CIRCLE":
+                return new Circle();
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:15]:
```suggestion
                return getRandomCircle();
```
lets add private methods to create each random figure, set all fields in those methods - not in constructors

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                return new RightTriangle();
+            case "ISOSCELES_TRAPEZOID":
+                return new IsoscelesTrapezoid();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("WHITE", 10);
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:30]: add constants for default values

File: src/main/java/core/basesyntax/Circle.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private int radius;
+
+    Circle() {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:6]: dont forget about access modifier

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Application.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.Random;
+
+public class Application {
+    private static final Random random = new Random();
+    private static final int figuresCount = 10;
+
+    public static void main(String[] args) {
+        int randomNumber = random.nextInt(figuresCount);
+        int number = 0;
+        FigureSupplier figureSupplier = new FigureSupplier();
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Application.java:12]: FigureSupplier can be a field of class

File: src/main/java/core/basesyntax/Application.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        for (int i = 0; i < randomNumber; i++) {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Application.java:13]: use one loop `for`

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Pay attention to common mistakes

File: src/main/java/core/basesyntax/Application.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+public class Application {
+    private static final Random RANDOM = new Random();
+    private static final int FIGURES_COUNT = 20;
+    private static final FigureSupplier FIGURE_SUPPLIER = new FigureSupplier();
+
+    public static void main(String[] args) {
+        int randomNumber = RANDOM.nextInt(FIGURES_COUNT);
+        int halfOfRandomNumber = randomNumber / 2;
+        for (int i = 0; i < randomNumber; i++) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:13]: would be more readable to use them in-place

File: src/main/java/core/basesyntax/Circle.java (hunk 3 above)
Comment (okuzan) [reply src/main/java/core/basesyntax/Circle.java:11]: not fixed, default isn't good option here

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+public class IsoscelesTrapezoid extends Figure {
+    private static final int NUMBER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    private double firstSide;
+    private double secondSide;
+    private int height;
+
+    IsoscelesTrapezoid() {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:14]: make constructors public everywhere

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class RightTriangle extends Figure {
+    private static final int NUMBER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/RightTriangle.java:8]: The same goes for ColorSupplier - it's not rightTriangle job to set itself a color. It just should be able to take the color from the constructor or setter and change it accordingly. The reason and nature of generation of these parameters are out of the scope of the class itself

**Review by okuzan — CHANGES_REQUESTED** [review]
Please go thru all of the comments. If you don't understand/agree with the suggested changes, ask in the chat.

**Review by okuzan — APPROVED** [review]
Well done 👍

**Pull request closed without merging**